TYPO3.Flow/Classes/TYPO3/Flow/Property/TypeConverter/DateTimeConverter.php

```diff
@@ -138,12 +139,14 @@ public function convertFrom($source, $targetType, array $convertedChildPropertie
             }
             if (isset($source['dateFormat']) && strlen($source['dateFormat']) > 0) {
                 $dateFormat = $source['dateFormat'];
+                $isFormatSpecified = true;
             }
         }
         if ($dateAsString === '') {
             return null;
         }
-        if (ctype_digit($dateAsString) && $configuration === null && (!is_array($source) || !isset($source['dateFormat']))) {
+        $isFormatConfigured = ($configuration !== null && $configuration->getConfigurationValue(DateTimeConverter::class, self::CONFIGURATION_DATE_FORMAT) !== null);
+        if (!$isFormatConfigured && !$isFormatSpecified && ctype_digit($dateAsString)) {
```

Comment:
well that's similar to how it is already, it's just that if there is configuration, but it hasn't defined dateFormat then it works similar to when there is no configuration.. which makes more sense than how it currently is IMO